app/models/compute_resources/foreman/model/vmware.rb

```diff
@@ -64,7 +64,7 @@ def datastores(opts = {})
       if opts[:storage_domain]
         name_sort(dc.datastores.get(opts[:storage_domain]))
       else
-        name_sort(dc.datastores.all(:accessible => true))
+        name_sort(dc.datastores.where(:accessible => true).all)
```

Comment:
These changes don't look like they'll work, as this is a Fog collection and not AR.  It doesn't have a `.where` method.